src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -207,9 +212,8 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
     val votesValidationState = validateVotes(fb.header)
     new ExtensionValidator(votesValidationState)
       .validateExtension(fb.extension, fb.header, lastExtensionOpt, lastHeaderOpt)
-      .validate(hdrHeight,
-        lastHeaderOpt.map(_.height + 1).getOrElse(ErgoHistory.GenesisHeight) == fb.header.height,
-        s"${fb.id} => ${lastHeaderOpt.map(_.height)} == ${fb.header.height}")
+      .validate(hdrHeight,lastHeaders.isEmpty || lastHeaders.head.height + 1 == fb.header.height,
+        s"${fb.id} => ${lastHeaderOpt.map(_.height + 1)} == ${fb.header.height}")
```

Comment:
Also description of logic would be helpful. also Why genesis check is no longer necessary.